Cabal/Distribution/Simple/Haddock.hs

```diff
@@ -312,59 +318,42 @@ fromLibrary verbosity tmp lbi lib clbi htmlTemplate haddockVersion = do
     ghcVersion <- maybe (die "Compiler has no GHC version")
                         return
                         (compilerCompatVersion GHC (compiler lbi))
+    return $ amendArgs ifaceArgs
+      { argGhcOptions = toFlag (opts, ghcVersion)
+      , argTargets    = inFiles
+      }
 
-    return ifaceArgs {
-      argHideModules = (mempty,otherModules $ bi),
-      argGhcOptions  = toFlag (opts, ghcVersion),
-      argTargets     = inFiles
-    }
+fromLibrary :: PackageDescription
+            -> Library
+            -> Verbosity
+            -> FilePath
+            -> LocalBuildInfo
+            -> ComponentLocalBuildInfo
+            -> Maybe PathTemplate
+            -> Version
+            -> IO HaddockArgs
+fromLibrary _ lib = fromTarget getLibSourceFiles amend (libBuildInfo lib) lib
   where
-    bi = libBuildInfo lib
+  amend args = args
+      { argHideModules = (mempty, otherModules $ libBuildInfo lib) }
 
-fromExecutable :: Verbosity
+fromExecutable :: PackageDescription
+               -> Executable
+               -> Verbosity
                -> FilePath
-               -> LocalBuildInfo -> Executable -> ComponentLocalBuildInfo
-               -> Maybe PathTemplate -- ^ template for HTML location
+               -> LocalBuildInfo
+               -> ComponentLocalBuildInfo
+               -> Maybe PathTemplate
                -> Version
                -> IO HaddockArgs
-fromExecutable verbosity tmp lbi exe clbi htmlTemplate haddockVersion = do
-    inFiles <- map snd `fmap` getExeSourceFiles lbi exe clbi
-    ifaceArgs <- getInterfaces verbosity lbi clbi htmlTemplate
-    let vanillaOpts = (componentGhcOptions normal lbi bi clbi (buildDir lbi)) {
```

Comment:
The rest of the comments are just stylistic fluff (and can be ignored), but this has got me worried. I don't see this code anywhere in the new diff. This code seems to be solving a very specific problem (interface file overwriting) and I don't see (1) any code in the new diff to handle it, or (2) a comment in the commit saying why this is not a problem anymore. It's possible it isn't (esp if the CI passes) but that is very non-obvious!